agent/xds/response/response.go

```diff
@@ -78,3 +78,12 @@ func MakeEnvoyRegexMatch(patt string) *envoy_matcher_v3.RegexMatcher {
 		Regex: patt,
 	}
 }
+
+func MakeEnvoyStringMatcher(patt string) *envoy_matcher_v3.StringMatcher {
+	return &envoy_matcher_v3.StringMatcher{
+		MatchPattern: &envoy_matcher_v3.StringMatcher_SafeRegex{
+			SafeRegex: MakeEnvoyRegexMatch(patt),
+		},
+		IgnoreCase: true,
```

Comment:
I don't see any other cases of `IgnoreCase` set to `true` in the repo. I think it's fine since it's only used for methods, but this will end up being a breaking change relative to v1.